resources/sdk/purecloudjava/templates/Logger.mustache

```diff
@@ -0,0 +1,410 @@
+                }
+            }
+            return logFormatValue;
+        } catch (IllegalArgumentException e) {
+            return null;
+        }
+    }
+    
+    public void trace(
+        String method,
+        String url,
+        Object requestBody,
+        int statusCode,
+        Map<String, String> requestHeaders,
+        Map<String, String> responseHeaders
+    ) {
+        LogStatement logStatement = new LogStatement(
+            new Date(),
+            "trace",
+            method,
+            url,
+            requestHeaders,
+            responseHeaders,
+            statusCode,
+            requestBodyToString(requestBody)
+        );
+        log(LogLevel.LTrace, logStatement);
+    }
+
+    public void debug(
+        String method,
+        String url,
+        Object requestBody,
+        int statusCode,
+        Map<String, String> requestHeaders
+    ) {
+        LogStatement logStatement = new LogStatement(
+            new Date(),
+            "debug",
+            method,
+            url,
+            requestHeaders,
+            statusCode,
+            requestBodyToString(requestBody)
+        );
+        log(LogLevel.LDebug, logStatement);
+    }
+    
+    public void error(
+        String method,
+        String url,
+        Object requestBody,
+        String responseBody,
+        int statusCode,
+        Map<String, String> requestHeaders,
+        Map<String, String> responseHeaders
+    ) {
+        LogStatement logStatement = new LogStatement(
+            new Date(),
+            "error",
+            method,
+            url,
+            requestHeaders,
+            responseHeaders,
+            statusCode,
+            requestBodyToString(requestBody),
+            responseBody
+        );
+        log(LogLevel.LError, logStatement);
+    }
+
+    private void log(LogLevel logLevel, LogStatement logStatement) {
+        if (logLevel.getOrder() >= this.level.getOrder()) {
+            String logString = logStatement.asString(this.format, this.logRequestBody, this.logResponseBody);
+            if (logToConsole)
+                System.out.println(logString);
+            if (this.bw != null) {
+                try {
+                    bw.write(logString);
+                    bw.newLine();
+                    bw.flush();
+                } catch (Exception e) {
+                    // no-op
+                }
+            }
+        }
+    }
+
+    private static String requestBodyToString(Object requestBody) {
+        if (requestBody != null) {
+            if (requestBody instanceof String) {
+                return requestBody.toString();
+            }
+            if (requestBody instanceof byte[]) {
+                return new String((byte[])requestBody, StandardCharsets.UTF_8);
+            }
+        }
+        return null;
+    }
+
+    public enum LogFormat {
+        JSON,
+        Text
+    }
+    
+    public enum LogLevel {
+        LNone(4),
```

Comment:
The levels don't seem to be correct. I'm seeing trace logs when it's set to debug. Debug should have debug and error.
I can't see any debug messages when it's set to trace, Trace should have trace, debug and error.
Can't see any error logs regardless of level.

Reply:
I corrected the log levels, seeing the error log now when I run the tests